app/src/main/java/com/a65apps/yuhnin/lesson1/presenters/ContactListPresenter.java

```diff
@@ -1,42 +1,45 @@
 package com.a65apps.yuhnin.lesson1.presenters;
 
-import android.os.Handler;
-import android.os.Looper;
-
 import androidx.annotation.NonNull;
 
-import com.a65apps.yuhnin.lesson1.callbacks.PersonListCallback;
-import com.a65apps.yuhnin.lesson1.pojo.PersonModelCompact;
+
 import com.a65apps.yuhnin.lesson1.repository.ContactRepository;
 import com.a65apps.yuhnin.lesson1.views.ContactListView;
 import com.arellomobile.mvp.InjectViewState;
 import com.arellomobile.mvp.MvpPresenter;
 
-import java.util.List;
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.disposables.CompositeDisposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
 
 @InjectViewState
-public class ContactListPresenter extends MvpPresenter<ContactListView> implements PersonListCallback {
+public class ContactListPresenter extends MvpPresenter<ContactListView> {
     @NonNull
-    private final ContactRepository contactRepository;
+    final ContactRepository contactRepository;
     @NonNull
-    private final Handler handler;
+    CompositeDisposable compositeDisposable;
 
     public ContactListPresenter(@NonNull ContactRepository contactRepository) {
         this.contactRepository = contactRepository;
-        handler = new Handler(Looper.getMainLooper());
+        this.compositeDisposable = new CompositeDisposable();
     }
 
     public void requestContactList(String searchString) {
-        contactRepository.getAllPersons(this, searchString);
+        compositeDisposable.add(contactRepository.getAllPersons(searchString)
```

Comment:
https://github.com/goblinr/AndroidLessonsCommonMistakes/blob/master/%D0%A0%D0%B5%D1%84%D0%B0%D0%BA%D1%82%D0%BE%D1%80%D0%B8%D0%BD%D0%B3_%D1%81%D1%83%D1%89%D0%B5%D1%81%D1%82%D0%B2%D1%83%D1%8E%D1%89%D0%B5%D0%B3%D0%BE_%D0%BF%D1%80%D0%B8%D0%BB%D0%BE%D0%B6%D0%B5%D0%BD%D0%B8%D1%8F_%D1%81_%D0%B8%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5%D0%BC_RxJava.md#%D0%BF%D0%BE%D1%81%D1%82%D0%BE%D1%8F%D0%BD%D0%BD%D0%BE-%D0%BA%D0%BE%D0%BF%D1%8F%D1%82%D1%81%D1%8F-%D0%BF%D0%BE%D0%B4%D0%BF%D0%B8%D1%81%D0%BA%D0%B8-%D0%B2-compositedisposable